### PR TITLE
`pkgutil` and Test for `corpus.py`

### PR DESCRIPTION
**Pull request by DavidHarrison**
Add a test for `corpus.py` as well as change it to use `pkgutil` to load the files which should allow loading the values even if the files aren't packaged as they are in the repo (e.g. in a standalone executable).

I'm happy to remove the `pkgutil` changes if you think they add unnecessary complexity.

### CODE REVIEW

File: sacremoses/corpus.py [hunk 1]
```diff
@@ -11,36 +11,34 @@ class Perluniprops:
     The files in the perluniprop.zip are extracted using the Unicode::Tussle
     module from http://search.cpan.org/~bdfoy/Unicode-Tussle-1.11/lib/Unicode/Tussle.pm
     """
-    def __init__(self):
-        self.datadir = os.path.dirname(os.path.abspath(__file__)) + '/data/perluniprops/'
```
Comment (alvations) [inline sacremoses/corpus.py]: It may be good to keep track of where the `datadir` is.

File: sacremoses/corpus.py [hunk 2]
```diff
@@ -11,36 +11,34 @@ class Perluniprops:
...
+    # These are categories similar to the Perl Unicode Properties
+    available_categories = ['Close_Punctuation', 'Currency_Symbol', 'IsAlnum',
+                            'IsAlpha', 'IsLower', 'IsN', 'IsSc', 'IsSo',
+                            'IsUpper', 'Line_Separator', 'Number',
+                            'Open_Punctuation', 'Punctuation', 'Separator',
+                            'Symbol', 'Lowercase_Letter', 'Titlecase_Letter',
+                            'Uppercase_Letter']
 
-    def chars(self, category=None, fileids=None):
+    def chars(self, category=None):
```
Comment (alvations) [inline sacremoses/corpus.py]: 👍

File: sacremoses/corpus.py [hunk 3]
```diff
@@ -99,10 +98,11 @@ def words(self, lang=None, ignore_lines_startswith='#'):
             filenames = ['nonbreaking_prefix.en']
 
         for filename in filenames:
-            with io.open(self.datadir+filename, encoding='utf8') as fin:
-                for line in fin:
-                    line = line.strip()
-                    if line and not line.startswith(ignore_lines_startswith):
-                        yield line
+            relative_path = os.path.join("data", "nonbreaking_prefixes", filename)
```
Comment (alvations) [inline sacremoses/corpus.py]: Really want to just stop support for Python 2.7 or lower version of Python 3 and use this instead: https://stackoverflow.com/a/44189428/610569 

But I think it's alright for now =)

**Comment by alvations** [comment]
@DavidHarrison hope you don't mind if I took the liberty to patch the unittest for Python2 support. If the changes looks good to you, I'll merge them. 

Thank you for helping to change the code to read the data using `pkgutil.get_data()` instead of the home-made I/O

**Comment by DavidHarrison** [comment]
@alvations Thank you so much for making the changes for Python 2, looks good to me!

**Comment by alvations** [comment]
Thanks @DavidHarrison !

**Pull request merged**